win32/glob.c

```diff
@@ -727,54 +671,62 @@ static int globextend(const Char *path, glob_t *pglob, size_t *limitp)
  * pattern matching function for filenames.  Each occurrence of the *
  * pattern causes a recursion level.
  */
-static int match(Char *name, Char *pat, Char *patend)
+static bool match(Char *name, Char *pat, Char *patend)
 {
-	int ok, negate_range;
+	bool ok, negate_range;
 	Char c, k;
 
 	while (pat < patend) {
 		c = *pat++;
 		switch (c & M_MASK) {
 		case M_ALL:
-			if (pat == patend)
-				return(1);
-			do
-				if (match(name, pat, patend))
-					return(1);
-			while (*name++ != EOS)
-				;
-			return(0);
+			if (pat == patend) {
+				return true;
+			}
+			do {
+				if (match(name, pat, patend)) {
+					return true;
+				}
+			} while (*name++ != EOS);
+			return false;
 		case M_ONE:
-			if (*name++ == EOS)
-				return(0);
+			if (*name++ == EOS) {
+				return false;
+			}
 			break;
 		case M_SET:
-			ok = 0;
-			if ((k = *name++) == EOS)
-				return(0);
-			if ((negate_range = ((*pat & M_MASK) == M_NOT)) != EOS)
+			ok = false;
+			if ((k = *name++) == EOS) {
+				return false;
+			}
+			if ((negate_range = ((*pat & M_MASK) == M_NOT)) != EOS) {
 				++pat;
+			}
 			while (((c = *pat++) & M_MASK) != M_END)
 				if ((*pat & M_MASK) == M_RNG) {
-					if (c <= k && k <= pat[1])
-						ok = 1;
+					if (c <= k && k <= pat[1]) {
+						ok = true;
+					}
 					pat += 2;
-				} else if (c == k)
-					ok = 1;
-			if (ok == negate_range)
-				return(0);
+				} else if (c == k) {
+					ok = true;
+				}
+			if (ok == negate_range) {
+				return false;
+			}
 			break;
 		default:
-			if (*name++ != c)
-				return(0);
+			if (*name++ != c) {
+				return false;
+			}
 			break;
 		}
 	}
-	return(*name == EOS);
+	return *name == EOS;
 }
 
 /* Free allocated data belonging to a glob_t structure. */
-PHPAPI void globfree(glob_t *pglob)
+PHP_WIN_GLOB void globfree(glob_t *pglob)
 {
 	register int i;
```

Comment:
nit: I question `register` usefulness wdyt ?

Reply:
Me too, but we should stick as close to upstream as possible. and need to update to latest ([where this appears to be gone](https://github.com/openbsd/src/blob/master/lib/libc/gen/glob.c); also it's no longer an `int`).

And see https://github.com/php/php-src/pull/15640/files#diff-8ad7174b356b2ed6831802d66a1e2ed4372cc20837f2346fae86d49297d40d36R973 ;)